src/test/java/io/netty/incubator/codec/http3/Http3ControlStreamOutboundHandlerTest.java

```diff
@@ -83,32 +97,75 @@ public void testGoAwayIdDecreaseWorks() throws Exception {
     public void testGoAwayIdIncreaseFails() throws Exception {
         // Let's mark the parent as inactive before we close as otherwise we will send a close frame.
         parent.close().get();
-        EmbeddedChannel channel = newStream(new Http3ControlStreamOutboundHandler(
-                true, settingsFrame, new ChannelInboundHandlerAdapter()));
-        assertTrue(channel.writeOutbound(new DefaultHttp3GoAwayFrame(4)));
-        ReferenceCountUtil.release(channel.readOutbound());
+        EmbeddedChannel channel = newStream(newHandler());
 
-        try {
-            channel.writeOutbound(new DefaultHttp3GoAwayFrame(8));
-            fail();
-        } catch (Exception e) {
-            assertException(Http3ErrorCode.H3_ID_ERROR, e);
+        if (server) {
+            writeValidFrame(channel, new DefaultHttp3GoAwayFrame(4));
+            writeInvalidFrame(Http3ErrorCode.H3_ID_ERROR, channel, new DefaultHttp3GoAwayFrame(8));
+        } else {
+            writeValidFrame(channel, new DefaultHttp3GoAwayFrame(1));
+            writeInvalidFrame(Http3ErrorCode.H3_ID_ERROR, channel, new DefaultHttp3GoAwayFrame(3));
         }
+
         assertFalse(channel.finish());
     }
 
     @Test
     public void testGoAwayIdUseInvalidId() throws Exception {
         parent.close().get();
         // Let's mark the parent as inactive before we close as otherwise we will send a close frame.
-        EmbeddedChannel channel = newStream(new Http3ControlStreamOutboundHandler(
-                true, settingsFrame, new ChannelInboundHandlerAdapter()));
-        try {
-            channel.writeOutbound(new DefaultHttp3GoAwayFrame(2));
-            fail();
-        } catch (Exception e) {
-            assertException(Http3ErrorCode.H3_ID_ERROR, e);
+        EmbeddedChannel channel = newStream(newHandler());
+
+        if (server) {
+            writeInvalidFrame(Http3ErrorCode.H3_ID_ERROR, channel, new DefaultHttp3GoAwayFrame(2));
         }
+
+        assertFalse(channel.finish());
+    }
+
+    @Test
+    public void testPassesUnknownFrame() throws Exception {
+        parent.close().get();
+        // Let's mark the parent as inactive before we close as otherwise we will send a close frame.
+        EmbeddedChannel channel = newStream(newHandler());
+
+        writeValidFrame(channel, new DefaultHttp3UnknownFrame(Http3CodecUtils.MIN_RESERVED_FRAME_TYPE,
+                Unpooled.buffer().writeLong(8)));
+
+        assertFalse(channel.finish());
+    }
+
+    @Test
+    public void testMaxPushIdFailsWhenReduced() throws Exception {
```

Comment:
Looks like this test and below can reuse some code

Reply:
after I reverted the code, doesn't seem like I can reuse something here